File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/io/StreamHandler.java
package ru.bozaro.gitlfs.client.io;

import javax.annotation.Nonnull;
import java.io.IOException;
import java.io.InputStream;

/**
 * Interface for handle stream of downloading data.
 *
 * @author Artem V. Navrotskiy
 */
@FunctionalInterface
public interface StreamHandler<T> {
  @Nonnull
  T accept(@Nonnull InputStream inputStream) throws IOException;
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/io/StreamHandler.kt
package ru.bozaro.gitlfs.client.io

import java.io.IOException
import java.io.InputStream

/**
 * Interface for handle stream of downloading ru.bozaro.gitlfs.common.data.
 *
 * @author Artem V. Navrotskiy
 */
fun interface StreamHandler<T> {
    @Throws(IOException::class)
    fun accept(inputStream: InputStream): T
}
